.ado/templates/apple-xcode-build-static-libs.yml

```diff
@@ -0,0 +1,63 @@
+#
+# Build the FluentUI React Native Test App in different configurations
+#
+steps:
+  # macOS Framework Debug
```

Comment:
Having all these param changes into the same template seems like a good candidate for a matrix

Reply:
Yeah.. I think I'll look into that for the next iteration and just turn this all info a separate job with a matrix. Wanted to get this PR out to get feedback first, I'll try and use a matrix here.

Reply:
Side note, since this is copied from FluentUI Apple. If I refactor this, there's gains we can have in that repo too. @rdeassis and @markavitale as an FYI :D

Reply:
Update @HeyImChris . Using a matrix sends the build to different agents, and I need the built libraries to all be on the same host machine afterwards so I can pack it. In my preiliminary testing, it didn't seem like I could split it out and I kept it all together